app/ui/browser/actions/main-actions.js

```diff
@@ -34,7 +34,16 @@ export function createTab(location: ?string = undefined,
     const id = uuid.v4();
 
     // Start loading a tab while asking the User Agent to start its browsing session.
-    dispatch({ type: types.CREATE_TAB, id, ancestorId, location, options, instrument: true });
+    // Dispatching a list ensures only one render and prevents racing to set focus across user
```

Comment:
@linclark so the re-render is purely an implementation detail; it's not *required*.  In general, I don't understand perfectly what our race story is across multiple action dispatches (assuming the actions are not thunks).  If we dispatched two actions, is there a guarantee that another action (switching tabs!) don't interleave?  I know things are synchronous mostly but that type of thing scares me.